src/main/resources/lib/nextjs/proxy.es6

```diff
@@ -150,15 +176,13 @@ function doRequest(originalReq, frontendRequestPath, xpSiteUrl, componentSubPath
         const xpSiteUrlWithoutEditMode = xpSiteUrl.replace(/\/edit\//, '/inline/');
 
         if (isHtml) {
-            response.body = renderSingleComponent
-                ? getSingleComponentHtml(response.body)
-                : response.body;
-
+            if (response.body && renderSingleComponent) {
+                response.body = getSingleComponentHtml(response.body);
+            }
             response.pageContributions = getPageContributionsWithBaseUrl(response, xpSiteUrlWithoutEditMode);
-
         }
 
-        if (isHtml || isJs || isCss) {
+        if (response.body && (isHtml || isJs || isCss)) {
```

Comment:
We replace all urls even in html and js? 😨 So we traverse every entire html js and css file that the sites generates on each request?!

Reply:
Yes for now, but we may start caching it later ;)